docs: Add comprehensive AuthBridge documentation (#1476)
kagenti/kagenti: docs/authbridge/deployment-guide.md

# AuthBridge Deployment Guide

## Deployment Modes

> AuthBridge supports two deployment modes. The proxy-sidecar mode is the recommended

Reviewer: ```suggestion
AuthBridge supports deployment modes `envoy-sidecar`, `waypoint`, and `proxy-sidecar`. The proxy-sidecar mode is the recommended
```

Author: Well, technically, we decided not to go forward with the `waypoint` mode, so it would be only these two.

1. The reverse proxy takes over the agent's original port (e.g., `:8000`)
2. The agent is moved to a free port (e.g., `:8001`) via `PORT` env var
3. `HTTP_PROXY` and `HTTPS_PROXY` env vars are injected into the agent container
4. The Kubernetes Service targetPort remains unchanged — traffic flows through the proxy

```yaml
# To use proxy-sidecar mode, annotate your workload:
metadata:
  labels:
    kagenti.io/type: agent
  annotations:
    kagenti.io/authbridge-mode: "proxy-sidecar"
```

### Traffic Flow

```
Inbound:   Client → Service:8000 → Reverse Proxy → Agent:8001
Outbound:  Agent → HTTP_PROXY → Forward Proxy → Token Exchange → External Service
```

## Envoy-Sidecar Mode (Advanced)

For environments that require transparent interception of all TCP traffic
(not just HTTP), the Envoy-sidecar mode uses iptables to redirect traffic:

```yaml
# Envoy-sidecar is the default if no annotation is set, but will change to
# proxy-sidecar in a future release
metadata:
  labels:
    kagenti.io/type: agent
  # No annotation = envoy-sidecar (current default)
```

This mode requires privileged mode for the iptables init container and adds

> Envoy as a dependency. Use it only when you need protocol-level transparent

Reviewer: ```suggestion
a sidecar running [Envoy](https://www.envoyproxy.io/docs/envoy/latest/) as a dependency. Use it only when you need protocol-level transparent
```

interception.

## Configuration

AuthBridge is configured via YAML with `${ENV_VAR}` expansion. The operator

> injects the configuration as a ConfigMap.

Reviewer: ```suggestion
mounts the configuration from a ConfigMap such as "authbridge-config-weather-service".
```

kubectl set env deployment/weather-agent -n team1 -c authbridge-proxy LOG_LEVEL=debug
```

### Runtime Log Toggle

Toggle between `info` and `debug` without restart by sending `SIGUSR1`:

```bash
kubectl exec deploy/weather-agent -n team1 -c authbridge-proxy -- \
  bash -c 'for f in /proc/[0-9]*/cmdline; do [ -r "$f" ] || continue; \
  c=$(<"$f"); [[ "$c" == /usr/local/bin/authbridge* ]] && \
  kill -USR1 "${f//[!0-9]/}" && break; done'
```

### What to Look For

| Log message | Meaning |
|---|---|
| `token-exchange: success` | Token exchanged successfully for outbound request |
| `token-exchange: failed` | Token exchange failed (check Keycloak connectivity) |
| `jwt-validation: rejected` | Inbound request failed JWT validation |
| `proxy: blocked host` | Outbound request to a disallowed destination |
| `a2a-parser: parsed message/send` | A2A protocol request parsed for guardrails |
| `mcp-parser: request` | MCP tool call intercepted and parsed |

## Troubleshooting

### Token Exchange Fails (503 from outbound calls)

1. Check Keycloak connectivity:
   ```bash
   kubectl exec deploy/weather-agent -n team1 -c authbridge-proxy -- \

>      curl -s http://keycloak.keycloak.svc:8080/realms/kagenti/.well-known/openid-configuration

Reviewer: - Our deployment is `deploy/weather-service`, not weather-agent
- Our sidecar is envoy-proxy, not authbridge-proxy
- Our envoy-proxy sidecar does not have the `curl` binary.
- Our Keycloak is at keycloak-service.keycloak.svc, not keycloak.keycloak.svc.
